checkpoint/util.go

```diff
@@ -109,22 +111,38 @@ func CopyFile(fs afero.Fs, src, dst string) error {
 	return rf.Copy(fs, srcf)
 }
 
-func httpToLocalFile(ctx context.Context, resource *url.URL, fs afero.Fs, dst string) error {
-	req, err := http.NewRequestWithContext(ctx, http.MethodGet, resource.String(), nil)
+func httpToLocalFile(
+	ctx context.Context,
+	resource *url.URL,
+	fs afero.Fs,
+	dst string,
+	retryMax int,
+	retryDelay time.Duration,
+) error {
+	c := retryablehttp.NewClient()
+	c.RetryMax = retryMax
+	c.RetryWaitMin = retryDelay
+	c.RetryWaitMax = retryDelay * 2
+	c.Backoff = retryablehttp.LinearJitterBackoff
+
+	req, err := retryablehttp.NewRequestWithContext(ctx, http.MethodGet, resource.String(), nil)
 	if err != nil {
 		return fmt.Errorf("create http request: %w", err)
 	}
-	resp, err := (&http.Client{}).Do(req)
-	urlErr := &url.Error{}
-	switch {
-	case errors.As(err, &urlErr):
-		return ErrCheckpointNotFound
-	case err != nil:
-		return fmt.Errorf("http get recovery file: %w", err)
+
+	resp, err := c.Do(req)
+	if err != nil {
+		// This shouldn't really happen. According to net/http docs for Do:
+		// "Any returned error will be of type *url.Error."
+		return fmt.Errorf("%w: %w", ErrCheckpointRequestFailed, err)
```

Comment:
I don't understand this comment. This does happen - we even have a test case for it that asserts that this specific error is returned?

Reply:
Sorry, stray comment not removed from an earlier version of the change, will fix

Reply:
Stray comment removed